tests/e2e/utils.py

```
@@ -164,6 +164,7 @@ async def msg_handler(
     disconnect: bool = False,
     force_error: bool = False,
     ignore_connection_closed: bool = False,
+    connection_attempts: list = None,
```

Ruff (0.15.15)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)